packages/common/src/store/pages/saved-page/selectors.ts

```diff
@@ -2,20 +2,149 @@ import { CommonState } from 'store/commonStore'
+  state.pages.savedPage.localRemovedAlbumReposts
+
+export const getLocalPlaylistFavorites = (state: CommonState) =>
+  state.pages.savedPage.localPlaylistFavorites
+export const getLocalPlaylistReposts = (state: CommonState) =>
+  state.pages.savedPage.localPlaylistReposts
+export const getLocalRemovedPlaylistFavorites = (state: CommonState) =>
+  state.pages.savedPage.localRemovedPlaylistFavorites
+export const getLocalRemovedPlaylistReposts = (state: CommonState) =>
+  state.pages.savedPage.localRemovedPlaylistResposts
+
+/** Get the tracks in currently selected category that have been added to the library in current session */
+export const getSelectedCategoryLocalAdds = (state: CommonState) => {
+  const selectedCategory = getSelectedCategory(state)
+  const localFavorites = getLocalTrackFavorites(state)
+  const localPurchases = getLocalTrackPurchases(state)
+  const localReposts = getLocalTrackReposts(state)
+  let localLibraryAdditions
+  if (selectedCategory === LibraryCategory.Favorite) {
+    localLibraryAdditions = localFavorites
+  } else if (selectedCategory === LibraryCategory.Purchase) {
+    localLibraryAdditions = localPurchases
+  } else if (selectedCategory === LibraryCategory.Repost) {
+    localLibraryAdditions = localReposts
+  } else {
+    // Category = ALL
+    localLibraryAdditions = {
+      ...localReposts,
+      ...localFavorites,
+      ...localPurchases
+    }
+  }
+
+  return localLibraryAdditions
+}
+
+const getSelectedCategoryLocalCollectionUpdates = (
+  state: CommonState,
+  props: { collectionType: 'album' | 'playlist'; updateType: 'add' | 'remove' }
+) => {
+  const { collectionType, updateType } = props
+  const selectedCategory = getSelectedCategory(state)
+  let localFavorites: ID[], localPurchases: ID[], localReposts: ID[]
+  if (updateType === 'add') {
+    localFavorites =
+      collectionType === 'album'
+        ? getLocalAlbumFavorites(state)
+        : getLocalPlaylistFavorites(state)
+    localPurchases =
+      collectionType === 'album' ? getLocalAlbumPurchases(state) : [] // Can't buy playlists
+    localReposts =
+      collectionType === 'album'
+        ? getLocalAlbumFavorites(state)
+        : getLocalPlaylistReposts(state)
+  } else {
+    localFavorites =
+      collectionType === 'album'
+        ? getLocalRemovedAlbumFavorites(state)
+        : getLocalRemovedPlaylistFavorites(state)
+    localPurchases = [] // Can't remove purchases
+    localReposts =
+      collectionType === 'album'
+        ? getLocalRemovedAlbumFavorites(state)
+        : getLocalRemovedPlaylistReposts(state)
+  }
+
+  if (selectedCategory === LibraryCategory.Favorite) {
+    return localFavorites
+  } else if (selectedCategory === LibraryCategory.Purchase) {
+    return localPurchases
+  } else if (selectedCategory === LibraryCategory.Repost) {
+    return localReposts
+  } else {
+    // Category = ALL
+    return Array.from(
+      new Set([...localReposts, ...localFavorites, ...localPurchases])
```

Comment:
Love the usage of Set here. Just confirming I understand correctly based on the docs for `Set`. It's going to use strict equal checking to compare items. So this would only filter out items that are references to the exact same object in memory. Is that the behavior we're going for?

I _think_ it works since collection ids are actually just `numbers` and the values we are putting in these arrays are just the IDs.

Reply:
Oh no, set is totally fine. I'm all about not adding lodash methods for things that are easy to do with built-ins :-)